### PR TITLE
Typescript issues

### PR DESCRIPTION
**Pull request by nvalois**
Missing those two props on CalendarStrip component. 

Moment is send on callback `onDateSelected` and `onWeekChanged` making errors if we do a `date.toDateString()` on the function.

### CODE REVIEW

**Comment by peacechen** [comment]
@nvalois 
I resolved conflicts from another PR. Does that look ok? You may want to review #235

File: index.d.ts [hunk 1]
```diff
@@ -75,15 +74,16 @@ declare module "react-native-calendar-strip" {
       scrollable?: boolean;
       startingDate?: Date;
       selectedDate?: Date;
-      onDateSelected?: (date: Date) => void;
-      onWeekChanged?: (start: Date, end: Date) => void;
-      onHeaderSelected?: (dates: {weekStartDate: Date, weekEndDate: Date}) => void;
+      onDateSelected?: ((date: Moment) => void);
+      onWeekChanged?: ((start: Moment, end: Moment) => void);
+      onHeaderSelected?: (({weekStartDate: Moment, weekEndDate: Moment}) => void);
```
Comment (sugersuger) [inline index.d.ts:79]: Let's change only the line like below.  

```tsx
onHeaderSelected?: ((dates: {weekStartDate: Moment, weekEndDate: Moment}) => void);
```

It fixes TypeScript linting-errors.

Comment (nvalois) [reply index.d.ts]: I've updated the PR with your feedback. I've also change the other `Date` to `Moment`, I was having an error when I was passing a Date for `datesWhitelist` with the function in CalendarDay.js `date.isBetween()` and `date.isSame()`

**Comment by peacechen** [comment]
@sugersuger 
This has been merged. Please check that the main branch works properly in Typescript projects. I resolved some more merge conflicts.

**Comment by sugersuger** [comment]
@peacechen 
OK, I will try.

**Pull request merged**